### ISSUE TITLE
question on tensor allocation

### ISSUE THREAD

**Issue by tsing90** [YudeWang/deeplabv3plus-pytorch#29]
Hi thanks for your implementation. 

I modified your code for my task, and I got an error about gpu resource allocation problem:
in your train.py, there is a line of code: `device = torch.device(0)` and `net.to(device)` ,
which means we used gpu0 to store the networks.

then for `labels_batched` and `labels_batched`:
you used `.to(1)` , which means we put tensors to gpu1 for loss calculation.

is that a problem due to two different gpus? because I got an error indicating that the tensor/weight/gradient for calculating CrossEntropyLoss are not from the same gpu, and they should moved to same gpu.

there will be no error if I change `.to(1)` to `.to(0)`. 

I'm using pytorch 0.4.1, and use more than 1 gpu. Much appreciated if you can have a look at my question. 

one more question, with one gpu, the per gpu memory usage for 4 images sized 512x512 input is 7.6G, but with multiple gpus, the per gpu memory usage for 4 inputs is around 10G, is that normal? it seemed data parallelization consumed more than 2G gpu ram.

**Comment by YudeWang** [comment]
Hi @tsing90,
I use `.to(1)` because the main gpu 0 always occupy more gpu ram than others. I have to move the loss calculation part to another gpu for memory saving. You can modify the code if gpu ram is enough.
For the second question, I am not sure about that, sorry.

**Comment by tsing90** [comment]
Thanks a lot for your reply. It will be grateful if I can ask one more question:
I would like to freeze the encoder part, and fine-tuning the decoder weights, how should I do? here is my code:
```
def freeze_encoder(model):
    for m in model.named_modules():
        if 'backbone' in m[0] or 'aspp' in m[0]:
            for p in m[1].parameters():
                p.require_grad = False
    return model
```
is that right? many thanks

**Comment by YudeWang** [comment]
@tsing90 That's ok, but make sure to use `net.module` for `nn.DataParallel`

**Comment by tsing90** [comment]
Thanks for your reply, have a good weekend ~
